pkg/server/server.go

```diff
@@ -66,11 +67,14 @@ func (j Joiner) Join(vat casm.Vat, r Router) (*Node, error) {
 		return nil, err
 	}
 
+	router := j.pubsub(vat.Logger, r)
+
 	vat.Export(host.Capability, host.Server{
-		ViewProvider:     c,
-		PubSubProvider:   j.pubsub(vat.Logger, r),
-		AnchorProvider:   j.anchor(),
-		DebugProvider:    j.Debugger.New(),
+		ViewProvider:      c,
+		PubSubProvider:    router,
+		AnchorProvider:    j.anchor(),
+		DebugProvider:     j.Debugger.New(),
+		DiscoveryProvider: j.service(router.PubSub()),
```

Comment:
This is a subtle thing, but passing in a `pubsub.Router` here means that all service discovery capabilities created by the provider will call out to the same underlying `capnp.Client`.  This in turn means that all their calls will be processed serially, one after another.

I think the correct way to handle this is actually to change the `Host` schema to take a `pubsub.Router` as an argument.  When you think about it, it's kind of nice.  Nothing in the OCAP security model says I have to get the pubsub and service-discovery caps from the same place. 

So, I would recommend changing `Host` schema to:

```capnp
service @4 (pubsub :import "pubsub.capnp".Router) -> (service :import "service.capnp".ServiceDiscovery);
```

Obviously, the provider interface will have to take an argument to its method now, too.